tests/test_client_app_coverage.py

```
@@ -0,0 +1,1634 @@
+        "fetch_raw_releases_data",
+        side_effect=requests.RequestException("network error"),
+    )
+    result = downloader.get_releases()
+    assert result == []
+
+
+def test_get_releases_pagination(downloader, mocker):
+    downloader.config["APP_VERSIONS_TO_KEEP"] = 50
+    call_count = [0]
+
+    def _fetch(params):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_fetch`

(ANN202)